src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -1815,8 +1815,10 @@ void CodeGen::genLclHeap(GenTree* tree)
             }
             else
             {
-                emit->emitLoadImmediate(EA_PTRSIZE, rsGetRsvdReg(), amount);
-                emit->emitIns_R_R_R(INS_sub, EA_PTRSIZE, REG_SPBASE, REG_SPBASE, rsGetRsvdReg());
+                if (tempReg == REG_NA)
+                    tempReg = tree->ExtractTempReg();
```

Comment:
> It doesn't really matter. It's allowed

The inverse is also allowed and it does ultimately matter. The end result comes down to area owner preference and while some people on the team are fine with it, other people on the team request that it not be done in areas they are regularly maintaining.

At the same time, something being allowed doesn't necessarily mean it is a good idea. There is always nuance and cases where it may be objectively better or worse than the alternative.

> and simple code takes less space.

This is absolutely not a definitive truth. Simple code *can* take less space. Equally, you can write complex or hard to read code that also takes less space.

It is frequently the case that adding whitespace, adding braces, adding comments, inserting additional named variables, and more can all make code simpler and easier to read.

Skipping out on brackets have been the root cause of several bugs, CVEs (security issues), and other problems across a multitude of code bases (including large enterprise ones). This can become particularly prevalent as a part of using macros, after doing a refactoring, etc. So while you might be able to save 2 lines by skipping braces, it is always safer to explicitly define them.

Reply:
I was under impression [CLR JIT Coding Conventions](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/clr-jit-coding-conventions.md#12.1) specifically enumerating cases for which braceless ifs are allowed leaves that to the contributor's preference.

But I can of course change it if area owners want braces here anyway.